### ISSUE TITLE
Fix api url for Github Enterprise

### ISSUE THREAD

**Issue by v2kiran** [microsoft/PowerShellForGitHub#117]
The api url has been coded in the following format: 
```powershell
$url = "https://api.$hostName/$UriFragment"
```

While this works for github it fails with the enterprise version because the api url is different:
```code
http(s)://[hostname]/api/v3
```

As a result when you call any of the function referencing Github enterprise you may get the following error:

```powershell
The remote name could not be resolved: 'api.myghhost.com'
At C:\Program 
Files\WindowsPowerShell\Modules\PowerShellForGitHub\0.8.0\GitHubCore.ps1:520 
char:9
+         throw $newLineOutput
+         ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The remote name.... 
   com':String) [], RuntimeException
    + FullyQualifiedErrorId : The remote name could not be resolved: 'api.myghhost.com'

```

Will raise a PR to fix this.

**Comment by HowardWolosky** [comment]
That sounds like a dupe of #101 (and the documentation for that feature can be found [here] (https://github.com/microsoft/PowerShellForGitHub/blob/master/README.md#GitHub-enterprise). Please re-activate with more details if I've misunderstood.

**Comment by HowardWolosky** [comment]
Actually, after you saying that not all enterprise installations have the API available at `api. <hostname>`?

**Comment by v2kiran** [comment]
hmm I thought all Github enterprise instances had  a different api url than the public github.com. This has been the case in my current setup as well as the last place I worked.

Are you saying that there are some github enterprise instances that are accessible at : `api.ghhost`?

[https://developer.github.com/enterprise/2.16/v3/#current-version](url)

**Comment by HowardWolosky** [comment]
That was the case for @Cellivar who implemented #101...  Reading the docs, it appears not to be the required case. Re-opening this issue. Feel free to provide a PR.

**Comment by Cellivar** [comment]
To my knowledge both API endpoints work with a strong preference from GitHub support for the api.<domain> endpoint. We have seen integrations using either one to no ill effect. 

If your deployment doesn't expose api.<domain> that's usually a sign of a configuration issue.
